cli/cmd/generate_gcp.go

```diff
@@ -0,0 +1,787 @@
+	QuestionGcpBucketLocation           = "Specify the Bucket Location: (optional)"
+	QuestionGcpBucketRetention          = "Specify the Bucket Retention Days: (optional)"
+	QuestionGcpBucketLifecycle          = "Specify the Bucket Lifecycle Rule Age: (optional)"
+	QuestionGcpEnableUBLA               = "Enable Uniform Bucket Level Access(UBLA)?"
+	QuestionGcpEnableBucketForceDestroy = "Enable Bucket Force Destroy?"
+	QuestionGcpUseExistingSink          = "Use an existing Sink?"
+	QuestionGcpExistingSinkName         = "Specify the existing Sink name"
+
+	GcpAdvancedOptIntegrationName      = "Customize Integration name(s)"
+	QuestionGcpConfigIntegrationName   = "Specify a custom Config integration name: (optional)"
+	QuestionGcpAuditLogIntegrationName = "Specify a custom AuditLog integration name: (optional)"
+
+	QuestionGcpAnotherAdvancedOpt      = "Configure another advanced integration option"
+	GcpAdvancedOptLocation             = "Customize output location"
+	QuestionGcpCustomizeOutputLocation = "Provide the location for the output to be written:"
+	GcpAdvancedOptDone                 = "Done"
+
+	// GcpRegionRegex regex used for validating region input
+	GcpRegionRegex = `(asia|australia|europe|northamerica|southamerica|us)-(central|(north|south)?(east|west)?)\d`
+
+	GenerateGcpCommandState                  = &gcp.GenerateGcpTfConfigurationArgs{}
+	GenerateGcpExistingServiceAccountDetails = &gcp.ExistingServiceAccountDetails{}
+	GenerateGcpCommandExtraState             = &GcpGenerateCommandExtraState{}
+	CachedGcpAssetIacParams                  = "iac-gcp-generate-params"
+	CachedAssetGcpExtraState                 = "iac-gcp-extra-state"
+
+	// gcp command is used to generate TF code for gcp
+	generateGcpTfCommand = &cobra.Command{
+		Use:   "gcp",
+		Short: "Generate and/or execute Terraform code for GCP integration",
+		Long: `Use this command to generate Terraform code for deploying Lacework into an GCP environment.
+
+By default, this command will function interactively, prompting for the required information to setup the new cloud account. In interactive mode, this command will:
+
+* Prompt for the required information to setup the integration
+* Generate new Terraform code using the inputs
+* Optionally, run the generated Terraform code:
+  * If Terraform is already installed, the version will be confirmed suitable for use
+	* If Terraform is not installed, or the version installed is not suitable, a new version will be installed into a temporary location
+	* Once Terraform is detected or installed, Terraform plan will be executed
+	* The command will prompt with the outcome of the plan and allow to view more details or continue with Terraform apply
+	* If confirmed, Terraform apply will be run, completing the setup of the cloud account
+
+This command can also be run in noninteractive mode. See help output for more details on supplying required values for generation.
```

Comment:
We need to cut these lines at a maximum of 80-90 chars per line, just so that it is easy to display in any terminal, for example:
```
$ lacework  configure -h
Configure settings that the Lacework CLI uses to interact with the Lacework
platform. These include your Lacework account, API access key and secret.

To create a set of API keys, log in to your Lacework account via WebUI and
navigate to Settings > API Keys and click + Create New. Enter a name for
the key and an optional description, then click Save. To get the secret key,
download the generated API key file.

Use the flag --json_file to preload the downloaded API key file.

If this command is run with no flags, the Lacework CLI will store all
settings under the default profile. The information in the default profile
is used any time you run a Lacework CLI command that doesn't explicitly
specify a profile to use.

You can configure multiple profiles by using the --profile flag. If a
config file does not exist (the default location is ~/.lacework.toml),
the Lacework CLI will create it for you.

Usage:
  lacework configure [flags]
  lacework configure [command]

Available Commands:
  list           List all configured profiles at ~/.lacework.toml
  show           Show current configuration data
  switch-profile Switch between configured profiles

Flags:
  -h, --help               help for configure
  -j, --json_file string   loads the API key JSON file downloaded from the WebUI

Global Flags:
  -a, --account string      account subdomain of URL (i.e. <ACCOUNT>.lacework.net)
  -k, --api_key string      access key id
  -s, --api_secret string   secret access key
      --api_token string    access token (replaces the use of api_key and api_secret)
      --debug               turn on debug logging
      --json                switch commands output from human-readable to json format
      --nocache             turn off caching
      --nocolor             turn off colors
      --noninteractive      turn off interactive mode (disable spinners, prompts, etc.)
      --organization        access organization level data sets (org admins only)
  -p, --profile string      switch between profiles configured at ~/.lacework.toml
      --subaccount string   sub-account name inside your organization (org admins only)

Use "lacework configure [command] --help" for more information about a command.
```

NOTE: AWS generate doesn't follow this rule, I am about to fix that.

Reply:
I can update the help text.
I think the command line flags help text is still going to push us over the line length though.

Reply:
Here is an updated copy of the help text with this line length roughly enforced.
I'm curious about how we can to police this going forward 🤔 
```
Use this command to generate Terraform code for deploying Lacework into an
GCP environment.

By default, this command will function interactively, prompting for the required
information to setup the new cloud account. In interactive mode, this command will:

* Prompt for the required information to setup the integration
* Generate new Terraform code using the inputs
* Optionally, run the generated Terraform code:
  * If Terraform is already installed, the version will be confirmed suitable for use
	* If Terraform is not installed, or the version installed is not suitable,
          a new version will be installed into a temporary location
	* Once Terraform is detected or installed, Terraform plan will be executed
	* The command will prompt with the outcome of the plan and allow to view more
          details or continue with Terraform apply
	* If confirmed, Terraform apply will be run, completing the setup of the
          cloud account

This command can also be run in noninteractive mode.
See help output for more details on supplying required values for generation.

Usage:
  lacework cloud-account iac-generate gcp [flags]

Flags:
      --apply                                         run terraform apply without executing plan or prompting
      --audit_log                                     enable audit log integration
      --audit_log_integration_name string             specify a custom audit log integration name
      --bucket_lifecycle_rule_age int                 specify the lifecycle rule age (default -1)
      --bucket_location string                        specify gcp bucket location
      --bucket_name string                            specify new bucket name
      --bucket_region string                          specify gcp bucket region
      --bucket_retention_days int                     specify the bucket retention days
      --config                                        enable config integration
      --config_integration_name string                specify a custom config integration name
      --enable_force_destroy_bucket                   enable force bucket destroy
      --enable_ubla                                   enable universal bucket level access(ubla)
      --existing_bucket_name string                   specify existing bucket name
      --existing_service_account_name string          specify existing service account name
      --existing_service_account_private_key string   specify existing service account key (base64 encoded)
      --existing_sink_name string                     specify existing sink name
  -h, --help                                          help for gcp
      --organization_id string                        specify the organization id (only set if organization_integration is set)
      --organization_integration                      enable organization integration
      --output string                                 location to write generated content
      --project_id string                             specify the project id to be used to provision lacework resources (required)
      --service_account_credentials string            specify a Service Account credentials JSON path (leave blank to make use of google credential ENV vars)

Global Flags:
  -a, --account string      account subdomain of URL (i.e. <ACCOUNT>.lacework.net)
  -k, --api_key string      access key id
  -s, --api_secret string   secret access key
      --api_token string    access token (replaces the use of api_key and api_secret)
      --debug               turn on debug logging
      --json                switch commands output from human-readable to json format
      --nocache             turn off caching
      --nocolor             turn off colors
      --noninteractive      turn off interactive mode (disable spinners, prompts, etc.)
      --organization        access organization level data sets (org admins only)
  -p, --profile string      switch between profiles configured at ~/.lacework.toml
      --subaccount string   sub-account name inside your organization (org admins only)
```

As mentioned, the CLI flags push us over the line length limit at the moment